server/api/crud/alerts.py

```diff
@@ -375,6 +385,20 @@ def reset_alert(self, session: sqlalchemy.orm.Session, project: str, name: str):
         self._get_alert_state_cached().cache_remove(session, alert.id)
         self._clear_alert_states(alert)
 
+    @staticmethod
+    def _should_reset_alert(old_alert_data, alert_data, force_reset):
+        if force_reset:
+            return True
+
+        # if one of these fields was modified then we should reset the alert
+        if (
+            old_alert_data.entities != alert_data.entities
+            or old_alert_data.trigger != alert_data.trigger
+            or old_alert_data.criteria != alert_data.criteria
+        ):
+            return True
+        return False
+
```

Comment:
Since EventEntities is a pydantic.BaseModel, it automatically includes an __eq__ method that compares all fields by default.
This means two EventEntities instances will be considered equal if all their attributes match

Reply:
> With some python magic 🪄 :
Nice!